x/dsmr/node_test.go

```diff
@@ -994,6 +1022,287 @@ func TestNode_NewBlock_IncludesChunkCerts(t *testing.T) {
 	}
 }
 
+// TestDuplicateChunksElimination tests that duplicate chunks that have appeared before are getting correctly eliminated.
+func TestDuplicateChunksElimination(t *testing.T) {
+	r := require.New(t)
+
+	node := newTestNode(t)
+
+	blk := Block{
+		ParentID: ids.GenerateTestID(),
+		Hght:     1,
+		Tmstmp:   1,
+		blkID:    ids.GenerateTestID(),
+	}
+	r.NoError(node.Accept(context.Background(), blk))
+
+	chunk, chunkCert, err := node.BuildChunk(
+		context.Background(),
+		[]tx{
+			{
+				ID:     ids.GenerateTestID(),
+				Expiry: 4,
+			},
+		},
+		4,
+		codec.Address{},
+	)
+	r.NoError(err)
+
+	blk = Block{
+		ParentID: ids.GenerateTestID(),
+		Hght:     2,
+		Tmstmp:   2,
+		blkID:    ids.GenerateTestID(),
+		ChunkCerts: []*ChunkCertificate{
+			&chunkCert,
+		},
+	}
+	r.NoError(node.Accept(context.Background(), blk))
+
+	r.NoError(node.storage.AddLocalChunkWithCert(chunk, &chunkCert))
```

Comment:
We shouldn't have to call this, since the chunk cert is already added locally once the chunk is finished being built from `node.BuildChunk`, and remote chunks are added within `Accept` if we don't have them ([ref](https://github.com/ava-labs/hypersdk/blob/8e83b6d7e28b2fd8487e40a4cf0b01b5b2d9de5f/x/dsmr/node_test.go#L1307)). 

I also think this is a weird way of writing the test because we're manually injecting the chunk into the database instead of testing how this would happen through the actual user-flow of dsmr. Maybe a better way would be to test the cases where you are calling `BuildChunk` a second time w/ the same args to make a duplicate chunk, and trying to build + accept the block? We should also test the case where a peer sends us a block referencing a duplicate chunk that we try to verify and fail.